File: web/src/app/Models/Dtos/Assesment/AssessmentViewDto.ts
import { QuestionResponseDto } from '../Question/question-response-dto';

export interface AssessmentViewDto {
  id: string;
  title: string;
  description: string;
  difficultyLevel: number;
  createdBy: string;
  createdAt: string;
  isActive: boolean;
  questions?: QuestionResponseDto[];
  questionIds?: string[];
}

File: backend/SkillProof/SkillProof.Entities/Dtos/Assesment/AssessmentViewDto.cs
using SkillProof.Entities.Dtos.Questions;
using SkillProof.Entities.Enums;

namespace SkillProof.Entities.Dtos.Assesment;

public class AssessmentViewDto
{
    public string Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DifficultyLevel DifficultyLevel { get; set; }
    public string CreatedBy { get; set; }
    public DateTime CreatedAt { get; set; }
    public bool IsActive { get; set; }
        
    public List<QuestionResponseDto> Questions { get; set; } = new List<QuestionResponseDto>();
    public List<string> QuestionIds { get; set; } = new List<string>();
}